sdk/dapp-kit/src/components/SuiClientProvider.tsx

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { SuiClient, getFullnodeUrl } from '@mysten/sui.js/client';
+import { SuiClientContext } from '../hooks/useSuiClient.js';
+import { useMemo } from 'react';
+
+export interface SuiClientProviderProps {
+	children: React.ReactNode;
+	client?: SuiClient;
+	url?: string;
+}
+
+export const SuiClientProvider = (props: SuiClientProviderProps) => {
+	const client = useMemo(
+		() =>
+			props.client ??
+			new SuiClient({
+				url: props.url ?? getFullnodeUrl('devnet'),
```

Comment:
I could also see making `client` a required prop here instead of falling back to this default

Reply:
yeah... this is something I am pretty unsure about.  Ideally what I want to be able to do is something like:

```ts
const { setNetwork, suiClient } = useSuiClientContext()

<NetworkSwitcher onChange={network => setNetwork(network.url) }  />
```

This would likely require constructing the clients dynamically 🤷 this will all likely change as more of this all gets implemented